.github/workflows/run_cli_on_repo_list.yml

```diff
@@ -0,0 +1,118 @@
+# This is a reusable workflow that runs the CLI on a provided list of repos, 
+# and uploads the JSONLD files along with the CLI log to a branch in neurobagel/openneuro-annotations.
+# The workflow updates the sha.txt file for repos on which the CLI ran successfully.
+name: Run CLI on repo list
+
+on:
+  workflow_call:
+    inputs:
+      # NOTE: If this input is empty, the workflow should not run
+      dataset-list:
+        description: "String representing a list of repos to update"
+        required: true
+        type: string
+
+jobs:
+  batch-run-cli:
+    name: Batch run CLI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        # NOTE: A token is needed if you want to git push to a repo
+        with:
+          ref: 'main'
+          token: ${{ secrets.ON_WF_PAT }}
+      
+      # git-annex is required for datalad
+      # Also setting author info here for Git to suppress datalad warnings
+      - name: Install git-annex
+        run: |
+          sudo apt-get update
+          sudo apt-get install git-annex
+          echo "git-annex version: $(git-annex version)"
+          git config --global user.name "GitHub Actions Bot"
+          git config --global user.email "<>"
+
+      # TODO: Cache Python dependencies
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r code/requirements.txt
+
+      - name: Run CLI
+        working-directory: code
+        # Passing input as a file to avoid issues with multiline strings
+        # NOTE: This step also removes the quotes in dataset-list (we add these back inside batch_run_cli.sh)
+        run: |
+          echo "${{ inputs.dataset-list }}" > dataset_list.txt
+          ./batch_run_cli.sh dataset_list.txt 2>&1 | tee -a LOG.txt
+      
+      # TODO: If this gets too slow, try https://github.com/actions/cache
+      - name: Upload JSONLD files as artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: jsonld-files
+          path: code/data/jsonld
+          # TODO: Add a retention policy to delete old artifacts
+
+      - name: Upload log file as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: cli-log-file
+          path: code/LOG.txt
+
+      - name: Commit and push updated sha.txt file
+        # If sha.txt was not updated (i.e., all datasets failed?), skip this step
+        run: |
+          if ! git diff --quiet code/sha.txt; then
```

Comment:
`! git diff --quiet` seems to return a non-zero exit code. Could that break the workflow?

Reply:
nope, that's the point!

`git diff --quiet` returns 0 if there are no modifications, or a non-zero code (1 usually, I think) if there are modifications. Since 0 is considered "true" in bash, we negate the condition to have it only evaluate to true when there are differences